### PR TITLE
test: add e2e tooling

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* adds node script for e2e commands
	* default generates test apps based on the domains in the suite (won't generate if enough exist already)
	* can rebuild with `--setup`, apply domain filter with `--domains`, and set the concurrency (how many test apps should run at once) with `--concurrency`
	* adds clean command to clean em all out
	* all other args are passed to `playwright`
* fixes the workflow 🤞🏼 

### Why is it needed?

* Means you can run multiple e2e suites in parallel 🪄 

### How to test it?

* give it a go

### Related Issues

* resolves CONTENT-1555

### TODO

- [ ] contributor documentation

### CODE REVIEW

**Comment by joshuaellis** [comment]
Hey all, hmu with feedback 🙇🏼‍♀️ when we're happy i'll update the contributor documentation. It's probably a bit rough round the edges but hopefully it's good to get us going 🏃🏼

File: packages/utils/typescript/package.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
   "dependencies": {
     "chalk": "4.1.2",
     "cli-table3": "0.6.2",
-    "fs-extra": "10.0.1",
+    "fs-extra": "10.0.0",
```
Comment (Marc-Roig) [inline packages/utils/typescript/package.json:35]: why the downgrade?

File: test/scripts/run-e2e-tests.js [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+               */
+              const pathToEnv = path.join(appPath, '.env');
+              const envFile = (await fs.readFile(pathToEnv)).toString();
+              const envWithoutPort = envFile.replace('PORT=1337', '');
+              await fs.writeFile(pathToEnv, envWithoutPort);
+              /**
+               * Always build! Just incase.
+               */
+              console.log(`building admin panel for test app at path: ${chalk.bold(appPath)}`);
+              await execa('yarn', ['strapi', 'build'], {
```
[the remark this answers is not in the document]
Comment (joshuaellis) [reply test/scripts/run-e2e-tests.js:119]: Doesn't look like it 👍🏼

**Review by Marc-Roig — APPROVED** [review]
this is amazing 🚀

File: packages/utils/typescript/package.json (hunk 1 above)
Comment (gu-stav) [reply packages/utils/typescript/package.json:35]: Good catch. This also means Dependebot can upgrade them together again.

**Pull request merged**